### PR TITLE
VoiceMessages: properly respect user's microphone choice

### PR DESCRIPTION
**Pull request by Masterjoona**
Use currently selected input device instead of the default one.
#2477

### CODE REVIEW

File: src/plugins/voiceMessages/utils.ts [hunk 1]
```diff
@@ -17,5 +17,7 @@
 */
 
 import { classNameFactory } from "@api/Styles";
+import { findByPropsLazy } from "@webpack";
 
+export const { getInputDeviceId } = findByPropsLazy("getInputDeviceId");
```
Comment (Vendicated) [inline src/plugins/voiceMessages/utils.ts:22]:
```suggestion
import { findStoreLazy } from "@webpack";

export const MediaEngineStore = findStoreLazy("MediaEngineStore");
```

File: src/plugins/voiceMessages/WebRecorder.tsx [hunk 2]
```diff
@@ -40,6 +41,7 @@ export const VoiceRecorderWeb: VoiceRecorder = ({ setAudioBlob, onRecordingChang
                 audio: {
                     echoCancellation: settings.store.echoCancellation,
                     noiseSuppression: settings.store.noiseSuppression,
+                    deviceId: getInputDeviceId()
```
Comment (Vendicated) [inline src/plugins/voiceMessages/WebRecorder.tsx:44]: this might return "default". are you sure that'll work?

Comment (Masterjoona) [reply src/plugins/voiceMessages/WebRecorder.tsx:44]: I tested it like a week ago and it worked fine

Comment (Vendicated) [reply src/plugins/voiceMessages/WebRecorder.tsx:44]: yep seems to work fine, at least for web

**Comment by Vendicated** [comment]
kiitos joona

**Pull request merged**